agent/engine/docker_task_engine_test.go

```diff
@@ -1513,3 +1513,91 @@ func TestCreateContainerOnAgentRestart(t *testing.T) {
 		t.Error("Unexpected error", metadata.Error)
 	}
 }
+
+// TestTaskUseExecutionRolePullECRImage tests the agent will use the execution role
```

Comment:
TestTaskUseExecutionRolePullECRImage tests that the agent will use the execution role credentials to pull from an ECR repository.